src/modules/seasons/seasons.controller.ts

```diff
@@ -1,12 +1,46 @@
-import { Controller, Get } from '@nestjs/common';
+import { 
+    Body, 
+    Controller, 
+    Get, 
+    Param, 
+    Post, 
+    UsePipes, 
+    ValidationPipe 
+} from '@nestjs/common';
+import { NotFoundException } from '@nestjs/common/exceptions';
 import { SeasonsService } from './seasons.service';
+import { CreateSeasonDto, FindOneSeasonParam, SeasonDto } from './seasons.types';
 
 @Controller('seasons')
 export class SeasonsController {
     constructor(private readonly seasonsService: SeasonsService) {}
 
     @Get()
-    getSeasons(): object {
-        return this.seasonsService.findAll();
+    async getSeasons(): Promise<SeasonDto[]> {
+        const res = await this.seasonsService.findAll();
+        if (res)
+            return res;
+        else
+            throw new NotFoundException();
+    }
+
+    @Get(':id')
+    @UsePipes(new ValidationPipe())
+    async getOneSeason(@Param() params: FindOneSeasonParam): Promise<SeasonDto | object> {
+        const res = await this.seasonsService.findOne(params.id);
+        if (res)
+            return res;
+        else
+            throw new NotFoundException();
+    }
```

Comment:
this is a bit nitpicky of me but I hate if/else statements

we want to add a little bit more description in the error as well...

```ts
const res = await this.seasonsService.findOne(params.id);
if (!res) {
  throw new NotFoundException(`no season with id=[${params.id}]`);
}

return res;
```